### ISSUE TITLE
Why use 'var' instead of a specific type?

### ISSUE THREAD

**Issue by LeoLeeTech** [1chz/realworld-java21-springboot3#6]
Isn't using a specific type clearer and more explicit

```java
    public Article save(Article article, Collection<Tag> tags) {
        var savedArticle = save(article);
        for (var tag : tagJpaRepository.saveAll(tags)) {
            savedArticle.addTag(new ArticleTag(savedArticle, tag));
        }
        return savedArticle;
    }
```

```java
    public Article save(Article article, Collection<Tag> tags) {
        Article savedArticle = save(article);
        for (Tag tag : tagJpaRepository.saveAll(tags)) {
            savedArticle.addTag(new ArticleTag(savedArticle, tag));
        }
        return savedArticle;
    }
```

**Comment by 1chz** [comment]
First of all, I mostly agree with your opinion. This project was simply a code kata to pass the time and explore new features of Java and Spring to their fullest.  

From my experience, using `var` for type inference has both pros and cons:  

1. When dealing with many variables, unifying all type declarations as `var` can make the code much cleaner and more readable. However, it's important to write code in a way that allows the variable type to be clearly understood from the declaration.  
2. Since `import` statements are not enforced, using `var` can reduce the amount of code that needs to be modified when making changes. Personally, I always start by reviewing dependencies through the `import` statements, and I found this aspect quite helpful.  
3. For especially complex generic types, replacing them with `var` can significantly improve code readability.  

In conclusion, this project had a highly experimental nature, and as a fairly conservative developer, I maintain a cautious stance on using `var`. I want you to know that I mostly agree with your perspective and share a similar view.

**Comment by LeoLeeTech** [comment]
OK, thank you very much!

I'm learning JPA using this project. I'm looking at every detail, so I might have a few questions to ask you.
